src/ipred.rs

```diff
@@ -1198,87 +1198,82 @@ unsafe fn filter_fn(
     p6: c_int,
 ) -> c_int {
     if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
-        *flt_ptr.offset(0) as c_int * p0
-            + *flt_ptr.offset(1) as c_int * p1
-            + *flt_ptr.offset(16) as c_int * p2
-            + *flt_ptr.offset(17) as c_int * p3
-            + *flt_ptr.offset(32) as c_int * p4
-            + *flt_ptr.offset(33) as c_int * p5
-            + *flt_ptr.offset(48) as c_int * p6
+        flt_ptr[0] as c_int * p0
+            + flt_ptr[1] as c_int * p1
+            + flt_ptr[16] as c_int * p2
+            + flt_ptr[17] as c_int * p3
+            + flt_ptr[32] as c_int * p4
+            + flt_ptr[33] as c_int * p5
+            + flt_ptr[48] as c_int * p6
```

Comment:
In this branch, the length of slice in the final loop is 48, which cannot be indexed at 48, is this a bug with dav1d?
https://godbolt.org/z/T4GhhnxfE

Reply:
I don't think this is an issue or we'd be seeing panics in the tests now that we're using safe slice indexing. In your example you're doing `flt_ptr = &flt_ptr[FLT_INC..]` before the call to `filter_fn`, which is incorrect. After the final loop iteration the length will be 48, but we don't pass `flt_ptr` to `filter_fn` after that point. See [this playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9e57720ec14d120877fc5569acbb55c2) for a demonstration of what happens. If you move the `dbg!` call to after doing `&flt_ptr[flt_incr..]` then it panics.

Reply:
> I don't think this is an issue or we'd be seeing panics in the tests now that we're using safe slice indexing. In your example you're doing `flt_ptr = &flt_ptr[FLT_INC..]` before the call to `filter_fn`, which is incorrect. After the final loop iteration the length will be 48, but we don't pass `flt_ptr` to `filter_fn` after that point. See [this playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9e57720ec14d120877fc5569acbb55c2) for a demonstration of what happens. If you move the `dbg!` call to after doing `&flt_ptr[flt_incr..]` then it panics.

Thanks for reply, it's my fault, I did not notice that it calls before slicing, which is fine.